### PR TITLE
feat: Consent Proof

### PR DESCRIPTION
**Pull request by alexrisch**
Added handling for consent proof in new conversation 
Updated protos

Closes out https://github.com/xmtp/xmtp-android/issues/220 after handling signature in invitation

### CODE REVIEW

File: library/src/main/java/org/xmtp/android/library/Conversations.kt [hunk 1]
```diff
@@ -148,6 +153,31 @@ data class Conversations(
...
     }
 
+    private fun validateConsentSignature(signature: String, clientAddress: String, peerAddress: String, timestamp: Long): Boolean {
+        val messageData = Signature.newBuilder().build().consentProofText(peerAddress, timestamp).toByteArray()
+        val signatureData = Numeric.hexStringToByteArray(signature)
+        val sig = Signature.parseFrom(signatureData)
+        val recoveredPublicKey = KeyUtil.recoverPublicKeyKeccak256(sig.rawData.toByteString().toByteArray(), Util.keccak256(messageData))
+            ?: return false
+        return clientAddress == KeyUtil.publicKeyToAddress(recoveredPublicKey)
+    }
```
Comment (nplasterer) [inline library/src/main/java/org/xmtp/android/library/Conversations.kt:163]: I kind of feel like this belongs in the KeyUtil class instead.

File: library/src/main/java/org/xmtp/android/library/Conversations.kt [hunk 2]
```diff
@@ -148,6 +153,31 @@ data class Conversations(
...
+
+    private fun handleConsentProof(consentProof: Invitation.ConsentProofPayload, peerAddress: String) {
+        val signature = consentProof.signature
+        val timestamp = consentProof.timestamp
+
+        if (!validateConsentSignature(signature, client.address, peerAddress, timestamp)) {
+            return
+        }
+        val contacts = client.contacts
+        runBlocking {
```
Comment (nplasterer) [inline library/src/main/java/org/xmtp/android/library/Conversations.kt:173]: Instead of blocking the main thread just make this function a suspend.

**Review by nplasterer — COMMENTED** [review]
This looks great just a small threading concern.

File: library/src/main/java/org/xmtp/android/library/Conversations.kt (hunk 1 above)
Comment (alexrisch) [reply library/src/main/java/org/xmtp/android/library/Conversations.kt]: Updated!

**Review by nplasterer — APPROVED** [review]

**Pull request merged**